### PR TITLE
fix #64: pipeline infrastructure — directory structure, pipeline.json, and artifact conventions

### LINKED ISSUE #64: Pipeline infrastructure: directory structure, pipeline.json, and artifact conventions
## Parent Issue

Part of #61 (Pipeline file-based handoff and clean-context invocability) — Sub-issue B.
Depends on: #63

## Description

Implement the foundational infrastructure for file-based pipeline handoff:

1. **Pipeline ID generation** — scan `docs/camel-kit/`, find max, increment, create directory. Format: `NNN-slug` (e.g., `001-order-processing`)
2. **Directory structure** — `docs/camel-kit/<PIPELINE_ID>/` with generic filenames (design-spec.md, implementation-plan.md, execution-report.md, validation-report.md)
3. **Pipeline state** — `.camel-kit/pipeline.json` with minimal `activePipeline` field
4. **Stage detection** — artifact presence determines pipeline stage (spec-kit pattern)
5. **Structured Rationale + Constraint fields** — enforce in design-assembly guide per component/EIP decision
6. **Structured iteration log** — `.camel-kit/verify-log.md` as audit trail for verify cycles

## Directory Structure

```
docs/camel-kit/
  001-order-processing/
    design-spec.md
    implementation-plan.md
    execution-report.md
    validation-report.md

.camel-kit/pipeline.json → {"activePipeline": "001-order-processing"}
```

## Acceptance Criteria

- [ ] Pipeline ID generation works (sequential NNN-slug)
- [ ] `docs/camel-kit/<PIPELINE_ID>/` directory created on pipeline start
- [ ] `.camel-kit/pipeline.json` created/updated with `activePipeline`
- [ ] Stage detection via artifact presence documented
- [ ] Rationale + Constraint fields added to design-assembly guide
- [ ] Verify iteration log structure defined
- [ ] Update documentation in `docs/` (architecture.md, commands.md, user-guide.md) to reflect new infrastructure
- [ ] `./mvnw clean install -B` passes

### PR DESCRIPTION
**Pull request by luigidemasi**
## Summary

Implements the foundational infrastructure for file-based pipeline handoff (part of #61):

- **`nextId` CLI command** — `camel-kit nextId <slug>` generates sequential pipeline IDs (`NNN-slug`) and creates `docs/camel-kit/<ID>/` directories
- **Shared pipeline guide** — `skills/shared/pipeline-infrastructure.md` documents conventions for directory layout, `pipeline.json`, stage detection, and verify-log format
- **Unified pipeline state** — `.camel-kit/pipeline.json` replaces `ship-state.json`, supporting both manual and ship-initiated modes
- **Rationale + Constraint fields** — design-assembly guide now enforces structured decision rationale per component/EIP
- **Verify iteration log** — `.camel-kit/verify-log.md` append-only audit trail for verify cycles
- **Skill wiring** — brainstorm, plan, execute, validate, verify, and ship all updated to use pipeline paths
- **Documentation** — architecture.md, commands.md, user-guide.md updated

## Test plan

- [ ] `./mvnw clean install -B` passes (verified locally)
- [ ] `NextIdCommandTest` — 5 tests covering sequential ID generation, directory creation, slug validation
- [ ] Verify `nextId` command appears in `camel-kit --help`
- [ ] Verify shared pipeline guide is included in JAR (`skills/shared/pipeline-infrastructure.md`)
- [ ] Review skill wiring: each pipeline skill references `shared/pipeline-infrastructure.md` and uses `docs/camel-kit/<PIPELINE_ID>/` paths

## Acceptance Criteria from Issue

- [x] Pipeline ID generation works (sequential NNN-slug)
- [x] `docs/camel-kit/<PIPELINE_ID>/` directory created on pipeline start
- [x] `.camel-kit/pipeline.json` created/updated with `activePipeline`
- [x] Stage detection via artifact presence documented
- [x] Rationale + Constraint fields added to design-assembly guide
- [x] Verify iteration log structure defined
- [x] Update documentation in `docs/` (architecture.md, commands.md, user-guide.md)
- [x] `./mvnw clean install -B` passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a camel-kit nextId CLI: validates slugs, generates the next sequential pipeline ID (zero-padded, supports >999), prints it, and creates the pipeline directory.

* **Pipeline**
  * Pipeline-scoped artifact layout under docs/camel-kit/<pipeline-id>, pipeline state persisted in .camel-kit/pipeline.json, and an append-only verify iteration audit log.

* **Documentation**
  * New pipeline infrastructure guide and updates to architecture, commands, user guide, templates, and skill docs.

* **Tests**
  * Expanded tests covering nextId behavior and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/luigidemasi/camel-kit/pull/67)

### CODE REVIEW

File: camel-kit-core/src/main/java/io/github/luigidemasi/camelkit/command/pipeline/NextIdCommand.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.stream.Stream;
+
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Parameters;
+
+@Command(name = "nextId", description = "Generate next sequential pipeline ID and create directory")
+public class NextIdCommand implements Runnable {
+
+    private static final Pattern ID_PATTERN = Pattern.compile("^(\\d{3})-.*$");
+    private static final Pattern SLUG_PATTERN = Pattern.compile("^[a-z0-9]+(-[a-z0-9]+)*$");
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply camel-kit-core/src/main/java/io/github/luigidemasi/camelkit/command/pipeline/NextIdCommand.java]: Fixed in 1fec4c0d. Changed `ID_PATTERN` from `\\d{3}` to `\\d+` to support IDs beyond 999. Added a boundary test (`handlesIdsBeyond999`) that verifies `999-last` + `1000-current` → `1001-next`.

File: camel-kit-core/src/main/java/io/github/luigidemasi/camelkit/command/pipeline/NextIdCommand.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    String slug;
+
+    Path baseDir = Path.of(".");
+
+    @Override
+    public void run() {
+        if (!SLUG_PATTERN.matcher(slug).matches()) {
+            System.err.println("Invalid slug: '" + slug
+                               + "'. Must be lowercase alphanumeric with hyphens (e.g. 'order-processing').");
+            return;
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply camel-kit-core/src/main/java/io/github/luigidemasi/camelkit/command/pipeline/NextIdCommand.java]: Fixed in 1fec4c0d. Error branches now throw `IllegalArgumentException` (for invalid slug) and `RuntimeException` (for I/O errors) instead of silently returning. Picocli translates uncaught exceptions to non-zero exit codes. Also added an `invalidSlugThrowsException` test.

File: camel-kit-core/src/main/resources/skills/camel-brainstorm/guides/design-assembly.md [hunk 3]
```diff
@@ -200,15 +209,25 @@ After assembling the spec, check:
...
 ---
 
 ## Save and Present
 
-1. Save the spec to `docs/design-spec.md` (greenfield) or `docs/migration-spec.md` (migration)
-2. Create `.camel-kit/config.properties` if it doesn't exist:
+1. Save the spec to `docs/camel-kit/<PIPELINE_ID>/design-spec.md` (both greenfield and migration)
+2. Create or update `.camel-kit/pipeline.json` with:
+   ```json
+   {
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply camel-kit-core/src/main/resources/skills/camel-brainstorm/guides/design-assembly.md:231]: Fixed in 1fec4c0d. Updated the presentation message from `docs/design-spec.md` to `docs/camel-kit/<PIPELINE_ID>/design-spec.md`.

File: camel-kit-core/src/main/resources/skills/camel-execute/SKILL.md [hunk 4]
```diff
@@ -172,7 +172,10 @@ Read `shared/iron-laws.md` for the full Iron Laws. This phase enforces ALL four:
 
 ### Step 1: Read Plan and Extract Tasks
 
-Read `docs/implementation-plan.md` (or the plan path specified).
+Resolve the active pipeline using `shared/pipeline-infrastructure.md`:
+1. Read `.camel-kit/pipeline.json` -> get `activePipeline`
+2. Read the plan from `docs/camel-kit/<activePipeline>/implementation-plan.md`
+3. The execution report will be saved to `docs/camel-kit/<activePipeline>/execution-report.md`
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply camel-kit-core/src/main/resources/skills/camel-execute/SKILL.md:179]: Fixed in 1fec4c0d. Updated Step 0 to use `docs/camel-kit/<activePipeline>/implementation-plan.md`.

File: camel-kit-core/src/main/resources/skills/camel-ship/SKILL.md [hunk 5]
```diff
@@ -35,14 +35,15 @@ Stage 0: BRAINSTORM  →  Stage 1: PLAN  →  Stage 2: EXECUTE  →  Stage 3: VA
 
 ### Before Starting
 
-1. Check for `--resume` flag. If set, read `.camel-kit/ship-state.json` and jump to `currentStage`.
-2. Check for `--start-from` flag. If set, verify prerequisite artifacts exist:
-   - `plan` requires `docs/design-spec.md`
-   - `execute` requires `docs/design-spec.md` AND `docs/implementation-plan.md`
+1. Generate pipeline ID: run `{COMMAND_PREFIX} nextId <slug>` (derive slug from input file name or user request).
+2. Check for `--resume` flag. If set, read `.camel-kit/pipeline.json`, verify `mode = "ship"`, and jump to `currentStage`.
+3. Check for `--start-from` flag. If set, verify prerequisite artifacts exist in `docs/camel-kit/<activePipeline>/`:
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply camel-kit-core/src/main/resources/skills/camel-ship/SKILL.md]: Fixed in 1fec4c0d. Reordered the Before Starting section: `--resume` check is now step 1 (with 'Skip steps 2-6' instruction), `nextId` is step 2 (only for new runs).

File: camel-kit-core/src/main/resources/skills/camel-validate/SKILL.md [hunk 6]
```diff
@@ -35,9 +47,8 @@ Provides the domain knowledge guides needed to validate generated Apache Camel r
 
 After completing all validation checks, generate a markdown report saved to:
 
-```
-docs/validation-report-YYYY-MM-DD_HH-mm.md
-```
+- **Pipeline mode:** `docs/camel-kit/<PIPELINE_ID>/validation-report.md`
+- **Standalone mode:** `docs/validation-report-YYYY-MM-DD_HH-mm.md`
 
 Use the current date and time for the filename (e.g., `validation-report-2026-04-22_14-30.md`).
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply camel-kit-core/src/main/resources/skills/camel-validate/SKILL.md:55]: Fixed in 1fec4c0d. Updated the HARD-RULE to distinguish pipeline mode (`docs/camel-kit/<PIPELINE_ID>/validation-report.md`) from standalone mode (timestamped in `docs/`).

File: camel-kit-core/src/main/resources/skills/shared/pipeline-infrastructure.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+## Directory Layout
+
+```
+docs/camel-kit/<PIPELINE_ID>/
+  design-spec.md           <- written by brainstorm
+  implementation-plan.md   <- written by plan
+  execution-report.md      <- written by execute
+  validation-report.md     <- written by validate
+  stamp-report.md          <- written by ship (stamp gate)
+```
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply camel-kit-core/src/main/resources/skills/shared/pipeline-infrastructure.md]: Fixed in 1fec4c0d. Added `text` language tag to the directory layout code fence.

File: docs/architecture.md [hunk 8]
```diff
@@ -593,7 +593,49 @@ The re-plan loop modifies affected TDD(s) only (never the BRD), max 3 rounds, wi
...
+### Directory Structure
+
+```
+docs/camel-kit/<PIPELINE_ID>/
+  design-spec.md           <- brainstorm output
+  implementation-plan.md   <- plan output
+  execution-report.md      <- execute output
+  validation-report.md     <- validate output
+  stamp-report.md          <- ship stamp gate output
+```
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply docs/architecture.md]: Fixed in 1fec4c0d. Added `bash` and `text` language tags to the two code fences.

File: docs/user-guide.md [hunk 9]
```diff
@@ -235,6 +235,30 @@ flowchart TB
...
+
+Each pipeline run creates a directory under `docs/camel-kit/` to persist artifacts across sessions:
+
+```
+docs/camel-kit/001-order-processing/
+  design-spec.md           <- brainstorm output (Phase 1)
+  implementation-plan.md   <- plan output (Phase 2)
+  execution-report.md      <- execute output (Phase 3)
+  validation-report.md     <- validate output (Phase 4)
+```
```
[the remark this answers is not in the document]
Comment (luigidemasi) [reply docs/user-guide.md]: Fixed in 1fec4c0d. Added `text` language tag to the directory listing code fence.
